pkg/driver/file/device_manager_file.go

```diff
@@ -680,6 +693,39 @@ func (d *DeviceManager) WriteLogs(m *logs.LogBundle) error {
 	return dev.AddLog(m)
 }
 
+// appExists return if an app has been created
+func (d *DeviceManager) appExists(u, instanceID uuid.UUID) bool {
+	_, err := os.Stat(d.getAppPath(u, instanceID))
+	if err != nil {
+		return false
+	}
+	if _, ok := d.devices[u]; !ok {
+		return false
+	}
+	return true
+}
+
+// WriteAppInstanceLogs write a message of AppInstanceLogBundle
+func (d *DeviceManager) WriteAppInstanceLogs(instanceID uuid.UUID, deviceID uuid.UUID, m *logs.AppInstanceLogBundle) error {
+	// make sure it is not nil
+	if m == nil {
+		return fmt.Errorf("invalid nil message")
+	}
+	// get the uuid
+	// check that the device actually exists
+	if !d.deviceExists(deviceID) {
+		return fmt.Errorf("unregistered device UUID: %s", deviceID)
+	}
+	if !d.appExists(deviceID, instanceID) {
+		d.devices[deviceID].AppLogs[instanceID] = &ManagedFile{
```

Comment:
I instantiate AppLogs here. And inside the same functions for the another managers.

Reply:
OK, I see it now. Makes sense. Thanks for clarifying.